### PR TITLE
Readonly & Disable were not working in TNC component

### PR DESCRIPTION
**Pull request by pavi41**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/site/js/termsandconditionsview.js [hunk 1]
```diff
@@ -62,10 +62,12 @@
                     threshold: [1],
                 })
                 function onIntersection ([{isIntersecting}]) {
+                    const isDisabled = self._model._jsonModel?.enabled && !self._model._jsonModel?.readOnly;
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/site/js/termsandconditionsview.js:65]: Please only use public API's don't access `_jsonModel.enabled`

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/site/js/termsandconditionsview.js [hunk 2]
```diff
@@ -62,10 +62,12 @@
...
                     if (isIntersecting) {
-                        self.children.filter(c => c.getModel()._jsonModel.fieldType === 'checkbox').forEach(cb => {
-                            cb.getModel().enabled = true;
-                        })
+                        if (isDisabled) {
+                              self.children.filter(c => c.getModel()._jsonModel.fieldType === 'checkbox').forEach(cb => {
+                                cb.getModel().enabled = true;
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/site/js/termsandconditionsview.js:69]: Why is this set to enabled true when TnC is disabled

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/site/js/termsandconditionsview.js (hunk 2 above)
Comment (pavi41) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/site/js/termsandconditionsview.js:69]: Rishi this is the logic to enable checkbox after reader has completely scrolled down  the message if TNC component is => enabled = true and readOnly=false

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/site/js/termsandconditionsview.js (hunk 1 above)
Comment (barshat7) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/site/js/termsandconditionsview.js:65]: The name of the variable 'isDisabled' seems to be a misleading. Shouldn't it be isEnabled?

Comment (pavi41) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/site/js/termsandconditionsview.js]: Changed!

Comment (pavi41) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/site/js/termsandconditionsview.js]: Implemented!

**Pull request merged**